### DISCUSSION TITLE
Is it possible to specify execution_options during a flush?

### DISCUSSION THREAD

**Discussion by sjhewitt** [sqlalchemy/sqlalchemy#10745] [Usage Questions] [answered]
I am using the `schema_translate_map` to enable multi-tenancy in my app.
For explicit queries, I'm able to use the `do_orm_execute` event to modify the execution options to set the `schema_translate_map`.
However, when a query is issued due to a flush, the `do_orm_execute` event is not triggered. I've looked at the `before_flush` event, however there's no way of modifying the execution options for queries that will be executed.
Is there any other event that I could use?

**Comment by zzzeek** [comment, marked as answer]
for an event approach you would use the [before_execute](https://docs.sqlalchemy.org/en/20/core/events.html#sqlalchemy.events.ConnectionEvents.before_execute) event, however for setting a translate map for either a `Session`, or a transaction in the `Session`, should be achieved directly:

```

# set schema translate map for a whole session
session = Session(bind=engine.execution_options(schema_translate_map={}))

# set schema translate map for a transaction

session = Session()
session.connection().execution_options(schema_translate_map={})

```

note also as mentioned in the warning at https://docs.sqlalchemy.org/en/20/core/connections.html#schema-translating you should only use a single schema translate map for a particular session, as you otherwise can have multiple objects with the same identity key that are from different schemas.
